src/components/Property/Types/SelectionProperty.tsx

```diff
@@ -1,30 +1,38 @@
 import { MultiSelect } from '@mantine/core';
 
-import { ConcretePropertyBaseProps } from '../types';
+import {
+  useGetPropertyDescription,
+  useGetSelectionPropertyValue,
+  usePropListeningState
+} from '@/api/hooks';
+import { PropertyProps } from '@/components/Property/types';
 
-interface Props extends ConcretePropertyBaseProps {
-  setPropertyValue: (newValue: string[]) => void;
-  additionalData: {
-    Options: string[];
-  };
-  value: string[];
-}
+export function SelectionProperty({ uri }: PropertyProps) {
+  const [value, setValue] = useGetSelectionPropertyValue(uri);
+  const { value: currentValue, setValue: setCurrentValue } = usePropListeningState<
+    string[] | undefined
+  >(value);
+
+  const description = useGetPropertyDescription(uri);
+
+  if (!value || !description || currentValue === undefined) {
+    return <></>;
+  }
+
+  const options = description.additionalData.Options as string[];
```

Comment:
Same here, make a type for this `additionalData` structure? Similarly to `AdditionalDataNumber` and `AdditionalDataVectorMatrix`?